### PR TITLE
[CDAP-12274] Shows all import errors on Add Entity modal instead of in Studio

### PR DESCRIPTION
**Pull request by tonybach**
In this PR (https://github.com/caskdata/cdap/pull/9757) that was merged for this JIRA (https://issues.cask.co/browse/CDAP-12274), when the user imports a non-JSON file from the Add Entity modal, we show the error on that modal. ~~However this behavior is not consistent with the other possible errors, which we all show in Studio. This PR removes the non-JSON import error from the Add Entity modal and shows it in Studio instead.~~ See my comment below.

### CODE REVIEW

File: cdap-ui/app/hydrator/controllers/create/create-studio-ctrl.js [hunk 1]
```diff
@@ -66,9 +66,9 @@ class HydratorPlusPlusStudioCtrl {
     }
 
     if ($stateParams.resourceCenterId) {
-      let inputFile = $window.localStorage.getItem($stateParams.resourceCenterId);
+      let inputFile = JSON.parse($window.localStorage.getItem($stateParams.resourceCenterId));
```
Comment (elfenheart) [inline cdap-ui/app/hydrator/controllers/create/create-studio-ctrl.js]: JSON.parse should be within a `try-catch` block. It's an unsafe operation

**Comment by tonybach** [comment]
Discussed with @ajainarayanan and @leacd, will add a commit to revert the above changes, and show all the errors on the Add Entity modal instead of showing them in Studio.

**EDIT**: Made the changes mentioned above. To do this I had to move `NonStorePipelineErrorFactory` and `GLOBALS` from Angular to React, then from the old Angular services just access these from the `window` object.

File: cdap-ui/app/cdap/services/PipelineErrorFactory.js [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+ * the License.
+*/
+
+import {objectQuery} from 'services/helpers';
+import GLOBALS from 'services/global-constants';
+
+let countUnFilledRequiredFields = (node) => {
+  var requiredFieldCount = 0;
+  if (typeof node._backendProperties === 'object' && Object.keys(node._backendProperties).length) {
+    // angular.forEach(node._backendProperties, (value, key) => {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/services/PipelineErrorFactory.js]: remove all commented out angular code?

**Review by elfenheart — APPROVED** [review]
👍  code LGTM. Need to rebase with develop

**Review by bdmogal — APPROVED** [review]

**Pull request merged**